editor/src/core/shared/element-template.ts

```diff
@@ -1137,8 +1201,42 @@ export function isJSXElement(element: JSXElementChild): element is JSXElement {
   return element.type === 'JSX_ELEMENT'
 }
 
+export function isJSExpressionValue(element: JSXElementChild): element is JSExpressionValue<any> {
+  return element.type === 'ATTRIBUTE_VALUE'
+}
+
+export function isJSExpressionNestedArray(
+  element: JSXElementChild,
+): element is JSExpressionNestedArray {
+  return element.type === 'ATTRIBUTE_NESTED_ARRAY'
+}
+
+export function isJSExpressionNestedObject(
+  element: JSXElementChild,
+): element is JSExpressionNestedObject {
+  return element.type === 'ATTRIBUTE_NESTED_OBJECT'
+}
+
+export function isJSExpressionFunctionCall(
+  element: JSXElementChild,
+): element is JSExpressionFunctionCall {
+  return element.type === 'ATTRIBUTE_FUNCTION_CALL'
+}
+
+export function isJSExpressionOtherJavaScript(
+  element: JSXElementChild,
+): element is JSExpressionOtherJavaScript {
+  return element.type === 'ATTRIBUTE_OTHER_JAVASCRIPT'
+}
+
 export function isJSXArbitraryBlock(element: JSXElementChild): element is JSXArbitraryBlock {
-  return element.type === 'JSX_ARBITRARY_BLOCK'
+  return (
+    isJSExpressionValue(element) ||
+    isJSExpressionNestedArray(element) ||
+    isJSExpressionNestedObject(element) ||
+    isJSExpressionFunctionCall(element) ||
+    isJSExpressionOtherJavaScript(element)
+  )
```

Comment:
I'm afraid without an exhaustive check here we will forget to extend this function. the only way I can imagine the exhaustive check though if we go with a helper function like `isTransientAction()` which enumerates _all_ possible JSXElementChild types and returns an explicit true or false.

Reply:
Yeah, this made sense at the time as I had those guards to hand, but it will definitely cause trip ups in the future.

Reply:
Fixed in 8ec7d8e.